### PR TITLE
Show incompatible error for missing platform URLs

### PR DESCRIPTION
**Pull request by kumar303**
Fixes https://github.com/mozilla/addons-frontend/issues/3266

Previously when an add-on declared itself compatible with a platform yet was missing a platform file, we just let the user try to install it. This resulted in a console error because there was no install URL. This patch shows an incompatible error banner and disables the install button instead.

### CODE REVIEW

File: src/core/utils.js [hunk 1]
```diff
@@ -417,6 +420,22 @@ export function isCompatibleWithUserAgent({
       return { compatible: false, reason: INCOMPATIBLE_NO_OPENSEARCH };
     }
 
+    // Even if an extension's version is marked compatible,
```
Comment (tofumatt) [inline src/core/utils.js]: These are the edge cases that make AMO so special 😆

File: tests/unit/amo/components/TestAddonCompatibilityError.js [hunk 2]
```diff
@@ -10,25 +10,30 @@ import {
...
 
 
 describe(__filename, () => {
-  const defaultUserAgentInfo = {
-    browser: { name: 'Firefox' },
-    os: { name: 'Plan 9' },
+  let store;
+
+  beforeEach(() => {
+    store = createStore().store;
```
Comment (tofumatt) [inline tests/unit/amo/components/TestAddonCompatibilityError.js]: This could use `dispatchClientMetadata` for a more accurate/complete "default" store.

File: tests/unit/amo/components/TestAddonCompatibilityError.js [hunk 3]
```diff
@@ -42,11 +47,10 @@ describe(__filename, () => {
   }
 
   it('renders a notice for non-Firefox browsers', () => {
-    const root = render({
-      lang: 'en-GB',
-      reason: INCOMPATIBLE_NOT_FIREFOX,
-      userAgentInfo: { browser: { name: 'Chrome' }, os: {} },
+    _dispatchClientMetadata({
```
Comment (tofumatt) [inline tests/unit/amo/components/TestAddonCompatibilityError.js]: Not sure why this was made when we could use the existing `dispatchClientMetadata` helper.

File: tests/unit/core/test_utils.js [hunk 4]
```diff
@@ -495,23 +526,72 @@ describe('isCompatibleWithUserAgent', () => {
...
+      UAParser(userAgentsByPlatform.windows.firefox40);
+
+    expect(isCompatibleWithUserAgent({ addon, userAgentInfo }))
+      .toEqual({
+        compatible: false,
+        reason: INCOMPATIBLE_UNSUPPORTED_PLATFORM,
+      });
+  });
+
+  it('allows non-extensions to have mismatching platform files', () => {
```
Comment (tofumatt) [inline tests/unit/core/test_utils.js]: Does this ever happen? I didn't think themes could/did set platform.

File: tests/unit/amo/components/TestAddonCompatibilityError.js (hunk 3 above)
Comment (kumar303) [reply tests/unit/amo/components/TestAddonCompatibilityError.js]: With this helper I don't have to pass `store` into the function every time

File: tests/unit/amo/components/TestAddonCompatibilityError.js (hunk 2 above)
Comment (kumar303) [reply tests/unit/amo/components/TestAddonCompatibilityError.js]: This component relies on `lang` and `userAgent` so I wanted each test to call `dispatchClientMetadata()` instead since the values are important. I can add a comment to this if it's helpful though.

File: tests/unit/core/test_utils.js (hunk 4 above)
Comment (kumar303) [reply tests/unit/core/test_utils.js]: I'm not sure but since this code is only working around a bug in extensions I wanted to limit it there until we know we need it elsewhere.
